### ISSUE TITLE
"Powered By" for Core?

### ISSUE THREAD

**Issue by TiddlyTweeter** [Jermolene/TiddlyWiki5#2953]
@Jermolene  Would it not be a good idea to have the **"Powered by TiddlyWiki"** in the core? 

Yes, with an option to switch it off. 

But it seems to me odd, that at such a low overhead, its not always in **every edition** as a default. 

We need more advertising & it seems a minimal obvious step. 

Many users would be quite happy it was there. Its pretty unobtrusive as badges go. Why do we bend over backwards NOT to ALWAYS signal origins?

**Comment by pmario** [comment]
Hmm, Way to obstrusive for me, especially on mobile screens.

**Comment by TiddlyTweeter** [comment]
Ciao PMario I never used TW on a small screen. I should give it a look. 

However, the evident problems on small screens I don't think entirely obviates my broader point, which is to **signal** more  about TW. Perhaps there are other ways to do that without spoiling your mobile experience?

**Comment by danielo515** [comment]
How do you think this will help to give me a broader audience ? I never saw a tw file that I didn't recognize as a tiddlywiki powered page.

**Comment by twMat** [comment]
> Would it not be a good idea to have the "Powered by TiddlyWiki" in the core?
>Yes, with an option to switch it off.

I take it you mean visible at front (e.g in "footer" or similar).

I agree with @TiddlyTweeter . It is not a coincidence that **all** big services you see do this. It is a matter of gaining users.

>obtrusive on mobile

Even if turned off?

>I never saw a tw file that I didn't recognize as a tiddlywiki powered page.

1) How do you know?
2) You (we) are not in the target group

...

My interpretation is that open source projects that are >10 years mature are either BIG or obscure. Less likely to be "in between". With that many years in the making, it's either caught on or not. If it hasn't, that does *not* mean it cannot, but it *does* mean "something is not done right" in terms of reaching the audience.

**Comment by TiddlyTweeter** [comment]
Right, @twMat, neither PMario, nor danielo515 are in the target group for promoting TW to. 

I find it bonkers that the Empty Edition has so little promotion of itself, or linking to its page (tiddlywiki.com).

Perhaps there is an alternative to "Powered By" ... How about this ...

_A default CAT icon on the sidebar that leads to an "About" Tiddler explaining what TW is, where to find it and where also to seek help. I can't see that messing up even the most fussy mobile users. On: hover shows "Powered by TiddlyWiki, click to learn more ..."_

Overall I think **we under-promote TW**. That helps nobody.

By way of background: Currently, AFAIK, there is no way of knowing if a "page" you are looking at on the web is a TW or not, unless its visually flagged so. Since we do NOT know how many TW are "in the wild" (we don't track) the issue, remains salient.

**Comment by TiddlyTweeter** [comment]
OVERALL: If you don't encourage new users to promote TW they won't. They will just assume you are not interested in promoting it. The Empty Edition gives that impression.

**Comment by pmario** [comment]
>OVERALL: If you don't encourage new users to promote TW they won't. They will just assume you are not interested in promoting it. The Empty Edition gives that impression.

I'm with you here. I'm ok with a footer with: "Powerd by TiddlyWiki" .. For me the "story area" is the only place, where it gets the visibility a promo link needs. Similar to what any other CMS does. 

I'm magine a footer, that is fixed at the bottom of the page and gets pushed down if the content gets longer. So users that don't care and just use the stuff as provided may keep it. 

The funny thing is .... If you search google for "[powered by footer](https://www.google.at/search?q=powered+by+footer&ie=utf-8&oe=utf-8&client=firefox-b&gfe_rd=cr&ei=HEqIWbz7IaiP8QeWiIGwCA)" ... all you find is "How can I remove the Powered by xyz" posts. 

**So if we want users to keep it, we need to add some extra value**, that they really keep it. "Powered by .... " imo isn't enough.

**Comment by danielo515** [comment]
I'm also ok about adding it to the empty edition (see the discussion about renaming empty to start or beginner edditi ) but not to the core. The idea of the core is to be lightweight and customizable, such banner is an unnecessary bloat.

Regards

**Comment by TiddlyTweeter** [comment]
I thought "Empty" was the core and ONLY the core?

**Comment by TiddlyTweeter** [comment]
@pmario ...

> So if we want users to keep it, we need to add some extra value ..."Powered by .... " imo isn't enough.

OK. 

Let's pretend. 

I am a user coming for first time. 

First up is the option to _"Customise My Footer"_ ... that adds your Name. A Copyright statement. Maybe a help badge. Maybe a way to create a simple columnised menu of links to key Tiddlers. 

That seems consistent with current designs.  Yes? And is Mobile Workable.

I agree that "Powered By ..." needs to take its place. In fact it will work better next to self-interest.

**Comment by danielo515** [comment]
Empty is an edition that ships the core , but the core alone is not usable, so the empty edition is a bit more than just the core . There is nothing forcing us to not include anything else in the base/empty edition without including it into the core

**Comment by danielo515** [comment]
I mean that you can include as many things as you want in the empty edition. That's why I proposed it to call it begginer or base edition rather than empty. At the end it's about user experience, and the empty edition is not very friendly for newcomers , and the worst thing is that an alternative does not exist officially

**Comment by TiddlyTweeter** [comment]
@danielo515 you need understand I am a naive user. When I view a TW downloaded as "empty" all I see is that its only plugin is the "core". I have NO idea what else is happening.

Frankly I can't get interested in what precisely is happening. ALL I know is that TW could promote itself better. And would benefit from that if it did that on ALL public releases, however constituted.

**Comment by danielo515** [comment]
I understand you position @TiddlyTweeter. But you have open the issue proposing to include such banner in the core, which I'm against. That's why I have exposed my point talking about the edition rather than core

**Comment by TiddlyTweeter** [comment]
@danielo515 this is likely a semantic issue. I do NOT care, really, if its in the "core" or not. 

**Just that its in public releases.** 

Excuse me for thinking it needs be in the core if it doesn't need to be. 

All I know is that is what it looks like. 

At the moment "empty" = "core". At least for naive users like me.

**Comment by tobibeer** [comment]
I think it's a good and sensible idea, that it should be a simple template for the bottom of the sidebar and with simple instructions for a user to turn it off, e.g. by removing the sidebar tag that will hopefully come with #2758, soon.

**Comment by TiddlyTweeter** [comment]
> @tobibeer: In fact, it possibly doesn't even have to be part of the core plugin, but just something you can simply delete.

I think that could work. I'm just slightly concerned we should not over-emphasise it can be deleted.

A related part of the issue is, perhaps, not so much the mechanism as encouraging generally the promotion of TW. 

Whilst it is true that developers nearly always respect and promote TW, and want to do so, they tend to do so in footnotes on plugins etc rather than is the more standard way of "up-front badges". 

All I was pointing to is the need for something, some flagging, in every edition that _"This is a TiddlyWiki & here is the link" that every user will see, sooner or later._

**Comment by tobibeer** [comment]
> I think that could work. I'm just slightly concerned we should not over-emphasise it can be deleted.

I think the overall philosophy always has been an emphasis on that it should not be hard.

> All I was pointing to is the need for something, some flagging, in every edition that "This is a TiddlyWiki & here is the link" that every user will see, sooner or later.

We have had this github style banner before and it was just too imposing in the end, I believe. So, the thing that might survive is a gentle note that speaks "Look, here's a mild hint at what this is. Interested?" ...not the literal text, but as an idea for how it should feel, from my point of view. Just as you discover TiddlyWiki, you discover this little pointer to... *♥ _TiddlyWiki_*.

**Comment by tobibeer** [comment]
Here's a little demo as to what I find just about enough promotion...

http://twlove.tiddlyspot.com

And here's what I find too much (sry @felixhayashi ;-):

http://tiddlymap.org

Notice the ribbon at the top right which was also features at tiddlywiki.com but not any more... and I think for a good reason.

**Comment by TiddlyTweeter** [comment]
>  @tobibeer  Here's a little demo as to what I find just about enough promotion...

Right. I agree. There. But not VISUALLY interfering.

**Comment by pmario** [comment]
@TiddlyTweeter -- Can be closed. There is a Powered By TiddlyWiki plugin in the core plugin library. Preview can be seen at: https://tiddlywiki.com/prerelease/#%24%3A%2Fplugins%2Ftiddlywiki%2Fpowered-by-tiddlywiki
